### ISSUE TITLE
Implement "My audit submissions" page

### ISSUE THREAD

**Issue by petermarks-gsa** [GSA-TTS/FAC#343]
## Acceptance criteria
(Acceptance criteria referring to login routing moved over to be part of #388 )

This issue only covers the page listing a user's audits, and doesn't address routing (or whether or not they should be redirected to their lone audit if they only have one).

### Frontend acceptance criteria
- [x]   The user will see a table containing the audits they're associated with.
- [x]   The table will include the following columns: **Report ID**, **Status**, **Auditee UEI**, **Fiscal Year End Date**, **Auditee Name**.
- [x]   The table will be sortable. The default sort order is **Report ID** ascending.
- [x]   The entries for **Report ID** will be links to the submissions. (URLs for submissions have not yet been implemented, but they should probably be something based on **Report ID**. Until we have a final route for that, the links should go to `/submission/<report_id>` or similar, even if those URLs aren't yet functional.)
- [x]   The header for **Auditee UEI** should have a link/tooltip/etc. for text explaining what a UEI is.
- [x]   There should be a “Start a new submission” form below the table. This consists of some text reminding the user not to duplicate already-existing submissions, a checkbox for agreeing to the terms and conditions, and a **Start a new submission** button.
- [x]   The page exhibits the appropriate behavior if the user clicks that button without checking the checkbox.
- [x]   Checking the checkbox and then clicking the button results in navigation to the first step of a new submission.

## Design
https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=2443%3A11490

## Tasks (already broken out in https://github.com/GSA-TTS/FAC/issues/391)
* 🅜 Back end - "My audit submissions" page
  * Surfaces a list of audits with Report id, Status, Auditee UEI, Fiscal year end date
  * Assume no sorting on backend or pagination (speculating that 400 is upper limit, most way less)
  * NEEDS CLARIFICATION: are we filtering available out of the list year
*   🅛 Backend: all audits associated with current user endpoint [#410](https://github.com/GSA-TTS/FAC/issues/410).
* 🅛 Front end - "My audit submissions" page
   * Sortable Columns: Report id, Status, Auditee UEI, Fiscal year end dat

## Subtasks in scope
* https://github.com/GSA-TTS/FAC/issues/391
* https://github.com/GSA-TTS/FAC/issues/424
* https://github.com/GSA-TTS/FAC/issues/410

## Related, but separate stories
* https://github.com/GSA-TTS/FAC/issues/388 / https://github.com/GSA-TTS/FAC/issues/401
* https://github.com/GSA-TTS/FAC/issues/319
* https://github.com/GSA-TTS/FAC/issues/212
* https://github.com/GSA-TTS/FAC/issues/342
* https://github.com/GSA-TTS/FAC/issues/332

## Not in scope
* Data table pagination.

**Comment by tadhg-ohiggins** [comment]
Backend overview:
*   We want the backend endpoint to return SACs
    *   Filtered by user!
        *   We need to handle auth.
        *   We need to get the user from the login/auth machinery.
        *   We need to get the submissions for that user, which will be looking for Access objects that have that user associated with them, and then looking for the SAC objects that have that user in any of these fields:
            *   certifying_auditee_contact
            *   certifying_auditor_contact
            *   auditee_contacts
            *   auditor_contacts

            OR we need to change the model so that many Access instances can point to a single SAC (but it's one-to-many and not many-to-many because each Access instance itself can only point to one SAC instance, and SAC instances wouldn't point to Access instances at all, only the reverse).
    
    *   Filtered by fiscal year of submission (maybe?)
    *   No pagination.
    *   No sorting.
    *   Create tests:
        *   Create a new user, hit endpoint, see no results.
        *   Create a new user, create an SAC (with Access for the user), see one result.
        *   Create a new user, create *n* SACs (each with Access for the user), see *n* results.
        *   Create a new user, create *n* SACs (each with Access with different roles for the user), see *n* results, each reflecting the correct role for a given SAC.
        *   Go through creation cycle, delete an SAC instance, verify that it's no longer listed.
        *   Go through creation cycle, delete an Access instance, verify that the corresponding SAC is no longer listed.
        *   … which means that to do this properly we need to have an endpoint for removing access. (Related to https://github.com/GSA-TTS/FAC/issues/333)

**Comment by rachelschneiderman** [comment]
Tasks:
- [ ] Empty `My audit submissions` page - Rachel (https://github.com/GSA-TTS/FAC-Frontend/pull/31)
- [ ] `My audit submissions` table
- [ ] Endpoint for retrieving my audit submissions

**Comment by petermarks-gsa** [comment]
@tadhg-ohiggins Per discussion, @heymatthenry and @jasnakai felt this story was sufficiently scoped in subtasks, so I've removed the "needs refinement" and "needs scoping" labels.

**Comment by tadhg-ohiggins** [comment]
Moving to blocked on the basis that until #399 and #391 are done, nobody will be working on this.

**Comment by JeanMarie-TTS** [comment]
@heymatthenry , I've created a starting version of the page and committed it in my branch. Can you work your magic on my)submissions.html please? The css needs to be fixed. In addition, the button at the end should be conditionall enabled when a check box is clicked - can you create the JS and the checkbox?

**Comment by melchoyce** [comment]
Just want to note this is the most up-to-date mockup: https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=4954%3A24137&t=TixnjgpGZuY87dAV-4
